src/core/src/node.cpp

```diff
@@ -707,8 +707,16 @@ bool ov::Node::constant_fold(OutputVector& output_values, const OutputVector& in
     bool all_constants = std::all_of(input_values.begin(), input_values.end(), [](const Output<Node>& input) {
         return ov::as_type_ptr<ov::op::v0::Constant>(input.get_node_shared_ptr());
     });
```

Comment:
@itikhono There is also an option to check for "can_be_folded" runtime info which is collected in scope of `pre_calculated_values_folding`. Not sure which one is better

Reply:
all_constants looks more solid now, let's leave it as is in this PR.
 I will double check if we can use `pre_calculated_values_folding` here.